specs/lightclient/sync-protocol.md

```diff
@@ -0,0 +1,181 @@
+# Minimal Light Client
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Introduction](#introduction)
+- [Constants](#constants)
+- [Configuration](#configuration)
+  - [Misc](#misc)
+  - [Time parameters](#time-parameters)
+- [Containers](#containers)
+    - [`LightClientSnapshot`](#lightclientsnapshot)
+    - [`LightClientUpdate`](#lightclientupdate)
+    - [`LightClientStore`](#lightclientstore)
+- [Light client state updates](#light-client-state-updates)
+    - [`is_valid_light_client_update`](#is_valid_light_client_update)
+    - [`apply_light_client_update`](#apply_light_client_update)
+    - [`process_light_client_update`](#process_light_client_update)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+Eth2 is designed to be light client friendly for constrained environments to access Eth2 with reasonable satefy and liveness. Such environments include resource-constrained devices (e.g. phones for trust-minimised wallets) and metered VMs (e.g. blockchain VMs for cross-chain bridges).
+
+This document suggests a minimal light client design for the beacon chain that uses sync committees introduced in [this beacon chain extension](./beacon-chain.md).
+
+## Constants
+
+| Name | Value |
+| - | - |
+| `FINALIZED_ROOT_INDEX` | `Index(BeaconState, 'finalized_checkpoint', 'root')` |
+| `NEXT_SYNC_COMMITTEE_INDEX` | `Index(BeaconState, 'next_sync_committee')` |
```

Comment:
Do these `Index` helpers equal `get_generalized_index` in https://github.com/ethereum/eth2.0-specs/blob/5f9112ad4227d12ea03c001517d53518e6e355f0/ssz/merkle-proofs.md?

If so, we may need to either make `merkle-proofs.md` executable again or implement it in `remerkleable`.

Reply:
> Do these Index helpers equal get_generalized_index

Yes :)

> we may need to either make merkle-proofs.md executable again or implement it in remerkleable

When we do so I suggest shortening "generalized_index" to something like "node_index", "tree_index", "Merkle_index", or just "index" for short.